envi/archs/arm/const.py

```diff
@@ -3,6 +3,32 @@
 MODE_JAZELLE    = 2
 MODE_THUMBEE    = 3
 
+'''
+bitfields to support different versions
+For simplicity will use 0xFFFF to say all.
+Note that thumb and arm are both represented
+thumbee is shown but may never be fully implimented
+since was already depreciated
+'''
+archBitMask = [
+#    architecture   bitmask              dec     hex
+    ('THUMB16',     0b0000000000000001), # 1      1
```

Comment:
Ok, I more or less added the thumb stuff mostly for testing purposes.
Using in the manner you mention makes it much easier to make modifications to it later.
  
 It also allowed me to test the thumb variants on their own but I suspect that just calling the ARM version with an address of negative value will do the same. 

Since ARMv7M is basically the same as ARMv7AR minus the ARM code, ie only THUMB2 code, should it be included with them?
I will work through this stuff and post an updated variation.

Date: Fri, 3 Jun 2016 05:39:13 -0700
From: notifications@github.com
To: vivisect@noreply.github.com
CC: redheadedrod@hotmail.com; author@noreply.github.com
Subject: Re: [atlas0fd00m/vivisect] Update unit tests for opcodes (#3)

In envi/archs/arm/const.py:

> @@ -3,6 +3,32 @@
>  MODE_JAZELLE    = 2
>  MODE_THUMBEE    = 3
>  
> +'''
> +bitfields to support different versions
> +For simplicity will use 0xFFFF to say all.
> +Note that thumb and arm are both represented
> +thumbee is shown but may never be fully implimented
> +since was already depreciated
> +'''
> +archBitMask = [
> +#    architecture   bitmask              dec     hex
> +    ('THUMB16',     0b0000000000000001), # 1      1

i believe thumb16, thumb2, and thumbee are all defined by the ARMv* items below.  for instance, ARMv5T includes THUMB16.  we should be able to clearly define an architecture using the ARMv* only.  for example, ARMv7M will be THUMB2 only.  ARMv7AR will support THUMB2.  if you'd like THUMB2 to be easily available, make it a bitfield combination of all ARMv* architectures which support THUMB2.


REV_ARMv7 = (REV_ARMv7M | REV_ARMv7AR)

REV_THUMB2 = (REV_ARMv6T2 | REV_ARMv7 | REV_ARMv8)


—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.